R/feature.align.R

```diff
@@ -89,119 +160,103 @@ feature.align <- function(features,
-                                         to_attach(thee[c(1, 2, 3, 4, 2, 6)], number_of_samples, use = "median"),
-                                         NA
-                        )
+                    gc() # call Garbage Collection for performance improvement?
+                # select a group
+                group_ids <- which(grps == sel.labels[i])
+                if (length(group_ids) > 1) {
+                    # select data from the group
+                    sample <- cbind(mz_values[group_ids], rt[group_ids], rt[group_ids], 
+                                    rt[group_ids], area[group_ids], sample_id[group_ids])
+                    # continue if data is from at least 'min_occurrence' samples
+                    if (validate_contents(sample, min_occurrence)) {
+                        return(select_mz(sample, mz_tol_relative, rt_tol_relative, min_occurrence, number_of_samples))
                     }
+                } else if (min_occurrence == 1) {
+                    sample_grouped <- c(mz_values[group_ids], rt[group_ids], rt[group_ids], 
+                                        rt[group_ids], area[group_ids], sample_id[group_ids])
+                    return(create_output(sample_grouped, number_of_samples, NA))
                 }
-                this.return
+                return(NULL)
             }
         
-        pk.times <- aligned.ftrs[, (5 + number_of_samples):(2 * (4 + number_of_samples))]
-        mz.sd.rec <- aligned.ftrs[, ncol(aligned.ftrs)]
-        aligned.ftrs <- aligned.ftrs[, 1:(4 + number_of_samples)]
+        # select columns: average of m/z, average of rt, min of m/z, max of m/z, median of rt per sample (the second to_attach call)
+        pk.times <- aligned_features[, (5 + number_of_samples):(2 * (4 + number_of_samples))]
+        mz_sd <- aligned_features[, ncol(aligned_features)]
+        # select columns: average of m/z, average of rt, min of m/z, max of m/z, sum of areas per sample (the first to_attach call)
+        aligned_features <- aligned_features[, 1:(4 + number_of_samples)]
         
-        colnames(aligned.ftrs) <-
+        # rename columns on both tables, samples are called "exp_i"
+        colnames(aligned_features) <-
             colnames(pk.times) <- c("mz", "time", "mz.min", "mz.max", paste("exp", 1:number_of_samples))
         
+        # return both tables and both computed tolerances
         rec <- new("list")
-        rec$aligned.ftrs <- aligned.ftrs
+        rec$aligned.ftrs <- aligned_features
         rec$pk.times <- pk.times
         rec$mz.tol <- mz_tol_relative
         rec$chr.tol <- rt_tol_relative
         
         if (do.plot) {
-            hist(mz.sd.rec, xlab = "m/z SD", ylab = "Frequency",
+            hist(mz_sd, xlab = "m/z SD", ylab = "Frequency",
                  main = "m/z SD distribution")
             hist(apply(pk.times[, -1:-4], 1, sd, na.rm = TRUE), 
                  xlab = "Retention time SD", ylab = "Frequency",
```

Comment:
can be moved to plot.R

Reply:
Do you have any suggestions? I think we would just create an artificial function which takes the same arguments as `hist`.